src/layouts/shared/NetworkErrorBoundary/networkErrorMetrics.ts

```diff
@@ -0,0 +1,47 @@
+import { metrics } from 'shared/utils/metrics'
+
+type SupportNetworkStatusCodes = 401 | 403 | 404 | 500
+
+export const sendNetworkErrorMetrics = (status: SupportNetworkStatusCodes) => {
+  if (status === 401) {
```

Comment:
idea: I think we can simplify this code quite a bit (and future-proof ourselves) with something like

return metrics.increment('network_errors.${status}`,1)

Reply:
Hrrm ... this has some type-safety implications ...

Reply:
currently if a network error code doesn't fall into line with any of this ... it does get lifted to a different error boundary

Reply:
hmmmmm, yeah I see what you mean.
I think the use case I was thinking was lets say someone accidentally forgets to add a conditional for a new error code we get it for "free" without having to do the extra leg work.

We could also use a switch case here with a default case to achieve the same if you'd be open to that?

Reply:
We sorta already have a catch all here